s3torchconnector/src/s3torchconnector/dcp/fsdp_filesystem.py

```diff
@@ -0,0 +1,249 @@
+            ValueError: If the mode is not 'rb' or 'wb'.
+        """
+        path = _path_to_str(path)
+        bucket, key = parse_s3_uri(path)
+
+        if mode == "wb":  # write mode
+            logger.debug("create_stream writable for %s", path)
+            with self._client.put_object(bucket, key) as stream:
+                yield stream
+        elif mode == "rb":  # read mode
+            logger.debug("create_stream readable for %s", path)
+            with self._client.get_object(bucket, key) as stream:
+                yield stream
+        else:
+            raise ValueError(
+                f"Invalid {mode=} mode argument: create_stream only supports rb (read mode) & wb (write mode)"
+            )
+
+    def concat_path(self, path: Union[str, os.PathLike], suffix: str) -> str:
+        """
+        Concatenate a suffix to the given path.
+
+        Args:
+            path (Union[str, os.PathLike]): The base path.
+            suffix (str): The suffix to concatenate.
+
+        Returns:
+            str: The concatenated path.
+        """
+        logger.debug("concat paths %s and %s", path, suffix)
+        path_str = os.fspath(path)
+        result = os.path.join(path_str, suffix)
+        return result
+
+    def init_path(self, path: Union[str, os.PathLike]) -> Union[str, os.PathLike]:
+        """
+        Initialize the path for the filesystem.
+
+        Args:
+            path (Union[str, os.PathLike]): The path to initialize.
+
+        Returns:
+            Union[str, os.PathLike]: The initialized path.
+        """
+        logger.debug("init_path for %s", path)
+        self._path = path
+        return self._path
+
+    def rename(
+        self, old_path: Union[str, os.PathLike], new_path: Union[str, os.PathLike]
+    ) -> None:
+        """Rename an object in S3.
+
+        This is emulated by copying it to a new path and deleting the old path. The deletion part is retried (see also
+        :func:`S3FileSystem._delete_with_retry`).
+
+        Args:
+            old_path (Union[str, os.PathLike]): The current path of the object.
+            new_path (Union[str, os.PathLike]): The new path for the object.
+
+        Raises:
+            ValueError: If the old and new paths point to different buckets.
+            S3Exception: If there is an error with the S3 client.
+        """
+        logger.debug("rename %s to %s", old_path, new_path)
+
+        old_path = _path_to_str(old_path)
+        new_path = _path_to_str(new_path)
+
+        old_bucket, old_key = parse_s3_uri(old_path)
+        new_bucket, new_key = parse_s3_uri(new_path)
+
+        if old_bucket != new_bucket:
+            raise ValueError(
+                f"Source and destination buckets cannot be different (rename does not support cross-buckets operations)"
+            )
+
+        self._client.copy_object(
+            src_bucket=old_bucket,
+            src_key=old_key,
+            dst_bucket=new_bucket,
+            dst_key=new_key,
+        )
```

Comment:
should we have some success log after the completion of each operation like copy and then delete? It can be easy for debugging.